File: security-platform/agents/dotnet/SecurityPlatform.Agent/PolicyResult.cs
namespace SecurityPlatform.Agent
{
    /// <summary>
    /// Result of a policy check.
    /// </summary>
    public class PolicyResult
    {
        public bool Allowed { get; }
        public string Reason { get; }

        public PolicyResult(bool allowed, string reason)
        {
            Allowed = allowed;
            Reason = reason;
        }
    }
}

File: security-platform/agents/java/src/main/java/com/securityplatform/agent/PolicyResult.java
package com.securityplatform.agent;

/**
 * Result of a policy check.
 */
public class PolicyResult {
    private boolean allowed;
    private String reason;

    public PolicyResult(boolean allowed, String reason) {
        this.allowed = allowed;
        this.reason = reason;
    }

    public boolean isAllowed() { return allowed; }
    public String getReason() { return reason; }
}
